lib/datadog/tracing/writer.rb

```diff
@@ -84,20 +104,28 @@ def stop
       # flush spans to the trace-agent, handles spans only
       # @!visibility private
       def send_spans(traces, transport)
-        return true if traces.empty?
+        return SendResult.new(status: SendResult::SUCCESS, retry_traces: []) if traces.empty?
```

Comment:
I could somewhat get behind the constant stuff as it can make the code easier to read and write / the list of possible values explicit and namespaced / avoid typos (making them fail loudly with a const missing).

But then we also can have valued union types for that.